src/openvpn/socket.c

```diff
@@ -1125,6 +1158,21 @@ create_socket(struct link_socket *sock, struct addrinfo *addr)
     }
     else if (addr->ai_protocol == IPPROTO_TCP || addr->ai_socktype == SOCK_STREAM)
     {
+#if defined(TARGET_LINUX) && defined(ENABLE_MPTCP)
+      if(sock->info.mptcp)
+      {
+	sock->sd = create_socket_mptcp(addr);
+	// Multipath TCP could fail because it is not enabled on this host
+	// Try regular TCP
+	if(sock->sd == -1)
+	{
+
+	  msg(M_NONFATAL, "Can't resolve MPTCP socket, fallback to TCP!");
+	  sock->sd = create_socket_tcp(addr);
```

Comment:
I guess the socket is created when starting OpenVPN, right? (I mean: I guess the client will not request the server to create an MPTCP socket, right?)

In this case, it means that the person who is launching OpenVPN has explicitly requested to use MPTCP: then better not to fallback in case of error, but fail, no? If yes, then it might be better to adapt `create_socket_tcp()` to support MPTCP, e.g. by adding a new parameter, or by setting `addr->ai_protocol = IPPROTO_MPTCP` is MPTCP has been requested (but that might have other consequences if `addr` is used later on, to be confirmed)

In my ideal world, MPTCP would be enabled by default, and used only if the client requested it. But I don't know if OpenVPN devs will accept that.

(while at it, but not directly related to ↑, you might need to do something else for DCO, because they don't support MPTCP yet: https://github.com/OpenVPN/ovpn-dco/issues/60 )